spacewalk/certs-tools/rhn_bootstrap_strings.py

```diff
@@ -310,41 +321,60 @@ def getRegistrationStackSh(saltEnabled):
 echo "-------------------------------------------------"
 
 function test_repo_exists() {{
-  local repourl="$CLIENT_REPO_URL"
-
-  $FETCH $repourl/repodata/repomd.xml
-  if [ ! -f "repomd.xml" ] ; then
-    echo "Bootstrap repo '$repourl' does not exist."
-    repourl=""
-    CLIENT_REPO_URL=""
-  fi
-  rm -f repomd.xml
+    local repourl="$CLIENT_REPO_URL"
+
+    $FETCH $repourl/repodata/repomd.xml
+    if [ ! -f "repomd.xml" ]; then
+        echo "Bootstrap repo '$repourl' does not exist."
+        repourl=""
+        CLIENT_REPO_URL=""
+    fi
+    rm -f repomd.xml
+}}
+
+function test_venv_enabled() {{
+    if [ $AVOID_VENV_SALT_MINION -ne 1 ]; then
+        local repourl="$CLIENT_REPO_URL"
+        if [ "$INSTALLER" == "zypper" ] || [ "$INSTALLER" == "yum" ]; then
+            ARCH=$(rpm --eval "%{{_host_cpu}}")
+        else
+            ARCH=$(dpkg --print-architecture)
+        fi
+        VENV_FILE="venv-enabled-$ARCH.txt"
+        $FETCH $repourl/$VENV_FILE
```

Comment:
The bootstrap repo is multiarch. This works only, if all architectures of a product has the venv or not.
Shipping venv for x86_64 but not for s390x would not work.
We need to check if this is the case.

Reply:
The package itself is arch dependant and in current state we are adding venv-salt-minion to the bootstrap repo almost manually and it's possible in this case that venv-salt-minion for x86_64 is in the repo, but aarch64 is not. I'm now trying to avoid such situation.

Reply:
I overlooked that the "arch" is in the filename. So you will have for every arch a dedicated file.
So it looks ok.